wiring/inc/spark_wiring_ble.h

```diff
@@ -0,0 +1,460 @@
+        uint8_t buf[BLE_MAX_ATTR_VALUE_PACKET_SIZE];
+        size_t len = sizeof(T) > BLE_MAX_ATTR_VALUE_PACKET_SIZE ? BLE_MAX_ATTR_VALUE_PACKET_SIZE : sizeof(T);
+        for (size_t i = 0, j = len - 1; i < len; i++, j--) {
+            buf[i] = reinterpret_cast<const uint8_t*>(&val)[j];
+        }
+        return setValue(buf, len);
+    }
+
+    void onDataReceived(OnDataReceivedCallback callback);
+
+    BleCharacteristicImpl* impl(void) const {
+        return impl_.get();
+    }
+
+private:
+    std::shared_ptr<BleCharacteristicImpl> impl_;
+    void construct(const char* desc, BleCharacteristicProperties properties, BleUuid& charUuid, BleUuid& svcUuid, OnDataReceivedCallback callback);
+};
+
+
+class BleService {
+public:
+    BleService();
+    BleService(const BleUuid& uuid);
+    ~BleService();
+
+    BleServiceImpl* impl(void) const {
+        return impl_.get();
+    }
+
+private:
+    std::shared_ptr<BleServiceImpl> impl_;
+};
+
+
+class BleScanResult {
+public:
+    BleAddress address;
+    BleAdvertisingData advertisingData;
+    BleAdvertisingData scanResponse;
+    int8_t rssi;
+};
+
+
+class BlePeerDevice {
+public:
+    BleDeviceRoles role;
+    BleAddress address;
+    BleConnectionParams connParams;
+    BleConnectionHandle connHandle;
+    int8_t rssi;
+
+    BlePeerDevice();
+    BlePeerDevice(const BleAddress& addr);
+    ~BlePeerDevice();
+
+    BleCharacteristic characteristic(const char* desc) const;
+    BleCharacteristic characteristic(const BleUuid& uuid) const;
+
+    bool connected(void) const;
+
+    BleGattServerImpl* gattsProxy(void) {
+        return gattsProxy_.get();
+    }
+
+    bool operator == (const BlePeerDevice& device);
+
+private:
+    std::shared_ptr<BleGattServerImpl> gattsProxy_;
+};
+
+
+class BleLocalDevice {
+public:
+    BleAddress address;
+
+    int on(void);
+    void off(void);
+
+    int setTxPower(int8_t val) const;
+    int8_t txPower(void) const;
+
+    int advertise(void);
+    int advertise(BleAdvertisingData* advertisingData, BleAdvertisingData* scanResponse=nullptr);
+    int advertise(uint16_t interval);
+    int advertise(uint16_t interval, BleAdvertisingData* advertisingData, BleAdvertisingData* scanResponse=nullptr);
+    int advertise(uint16_t interval, uint16_t timeout);
+    int advertise(uint16_t interval, uint16_t timeout, BleAdvertisingData* advertisingData, BleAdvertisingData* scanResponse=nullptr);
+    int advertise(const BleAdvertisingParams& params);
+    int advertise(const BleAdvertisingParams& params, BleAdvertisingData* advertisingData, BleAdvertisingData* scanResponse=nullptr);
+
+    int advertise(const iBeacon& iBeacon, bool connectable=false);
+    int advertise(uint16_t interval, const iBeacon& iBeacon, bool connectable=false);
+    int advertise(uint16_t interval, uint16_t timeout, const iBeacon& iBeacon, bool connectable=false);
+    int advertise(const BleAdvertisingParams& params, const iBeacon& iBeacon, bool connectable=false);
```

Comment:
This applies to all `advertise()` methods:
1. Time arguments should probably be `system_tick_t`
2. If the argument is a pointer and is not going to be modified by the function, it should probably be `const`
3. Nitpicking: spaces around `=`

Reply:
1. BLE spec requires 2-octets, while `system_tick_t` is 4-octets.
2 and 3. Done.